lib/presentation/profile/profile_view.dart

```diff
@@ -57,36 +57,139 @@ class ProfileScreen extends StatelessWidget {
+                                          content: Text('Coming soon!'),
+                                          duration: Duration(milliseconds: 500),
+                                        );
+                                        ScaffoldMessenger.of(context)
+                                            .showSnackBar(snackBar);
+                                      },
+                                      horizontalPadding: 10,
+                                      width: 115,
+                                    ),
+                                  ],
                                 ),
+                                10.toVerticalSizedBox,
+                                if (state.hostelChangeStatus['detail'] !=
+                                        null ||
+                                    (state.hostelChangeStatus['is_approved'] !=
+                                            null &&
+                                        state
+                                            .hostelChangeStatus['is_approved']))
+                                  ProfileTextButton(
+                                    title: 'Request for Hostel Change',
+                                    onPressed: () {
+                                      context.router.push(HostelChangeRoute());
+                                    },
+                                    horizontalPadding: 50,
+                                    width: 248,
+                                  ),
+                                if (state.hostelChangeStatus['detail'] == null)
+                                  TextButton(
+                                    onPressed: () => {
+                                      showDialog(
+                                          context: context,
+                                          builder: (ctx) => AlertDialog(
+                                                title: const Text(
+                                                    'Cancel Hostel Change Request'),
+                                                content: const Text(
+                                                  'You have already requested for a hostel change. Do you want to cancel it?',
+                                                  textAlign: TextAlign.center,
+                                                ),
+                                                actions: [
+                                                  TextButton(
+                                                    onPressed: () {
+                                                      context
+                                                          .read<
+                                                              ProfilePageBloc>()
+                                                          .add(
+                                                              const DeleteHostelChangeRequest());
+                                                      Navigator.pop(ctx);
```

Comment:
Again don't pollute the codebase. We are using `context.router.pop`. Be consistent!!